ovs-service/src/main/java/org/dcsa/ovs/service/VesselScheduleService.java

```diff
@@ -1,21 +1,172 @@
+import org.dcsa.ovs.transferobjects.*;
+import org.dcsa.ovs.transferobjects.enums.DimensionUnit;
+import org.dcsa.ovs.transferobjects.enums.PortCallStatusCode;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.Objects;
 
 @Service
 @RequiredArgsConstructor
 public class VesselScheduleService {
-  private final VesselScheduleRepository repository;
-  private final VesselScheduleMapper mapper;
 
-  public List<VesselScheduleTO> findAll() {
-    return ListMapper.convertList(repository.findAll(), mapper::toTO);
+  private final ServiceRepository serviceRepository;
+
+  @Builder
+  public static class ServiceSchedulesFilters {
+    String carrierServiceCode;
+    String universalServiceReference;
+    String vesselIMONumber;
+    String vesselName;
+    String voyageNumber;
+    String universalVoyageReference;
+    String unLocationCode;
+    String facilitySMDGCode;
+    String startDate;
+    String endDate;
+    Integer limit;
+    String cursor;
+    String apiVersion;
+  }
+
+  public List<VesselScheduleTO> findAll(ServiceSchedulesFilters requestFilters) {
+
+    return serviceRepository.findAll().stream()
+        .map(
+            service ->
+                VesselScheduleTO.builder()
+                    .carrierServiceName(service.getCarrierServiceName())
+                    .carrierServiceCode(service.getCarrierServiceCode())
+                    .universalServiceReference(service.getUniversalServiceReference())
+                    .vessels(extractVesselsToVesselTO(service.getVessels()))
+                    .build())
+        .toList();
+  }
+
+  private List<VesselTO> extractVesselsToVesselTO(List<Vessel> vessels) {
+    return vessels.stream()
+        .map(
+            vessel ->
+                VesselTO.builder()
+                    .vesselName(vessel.getName())
+                    .vesselIMONumber(vessel.getImoNumber())
+                    .vesselOperatorCarrierSMDGCode(vessel.getOperatorCarrier().getSmdgCode())
+                    .vesselCallSign(vessel.getCallSign())
+                    .vesselLength(vessel.getLength())
+                    .vesselWidth(vessel.getWidth())
+                    .dimensionUnit(
+                        DimensionUnit.valueFromString(vessel.getDimensionUnit()).orElse(null))
+                    .isDummyVessel(vessel.getIsDummy())
+                    .portCalls(extractTransportCallToPortCallTo(vessel.getPortCalls()))
+                    .build())
+        .toList();
+  }
+
+  private List<PortCallTO> extractTransportCallToPortCallTo(List<TransportCall> transportCalls) {
+    return transportCalls.stream()
+        .map(
+            transportCall ->
+                PortCallTO.builder()
+                    .transportCallReference(transportCall.getReference())
+                    .importVoyageNumber(
+                        Objects.requireNonNullElse(transportCall.getImportVoyage(), new Voyage())
```

Comment:
I'd prefer either make a "Voyage emptyVoyage = new Voyage();" in the start of the method or in the service and use that or use
Objects.requireNonNullElseGet(transportCall.getImportVoyage(), Voyage::new)

To avoid spawning a new Voyage object for every field

Reply:
Incidentally in kotlin this would simply be
.importVoyageNumber(transportCall.importVoyage?.carrierVoyageNumber)

the ?. operator would do the null-check for you